client/jetpack-cloud/sections/partner-portal/primary/payment-method-list/index.tsx

```diff
@@ -12,13 +13,112 @@ import Main from 'calypso/components/main';
 import CardHeading from 'calypso/components/card-heading';
 import DocumentHead from 'calypso/components/data/document-head';
 import SidebarNavigation from 'calypso/jetpack-cloud/sections/partner-portal/sidebar-navigation';
-import { Button } from '@automattic/components';
+import { getAllStoredCards } from 'calypso/state/stored-cards/selectors';
+import QueryStoredCards from 'calypso/components/data/query-stored-cards';
+import {
+	VisaLogo,
+	MastercardLogo,
+	AmexLogo,
+} from 'calypso/my-sites/checkout/composite-checkout/components/payment-logos';
+import EllipsisMenu from 'calypso/components/ellipsis-menu';
+import PopoverMenuItem from 'calypso/components/popover/menu-item';
+import Gridicon from 'calypso/components/gridicon';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+function StoredCard( props ): ReactElement {
+	let logo;
+	const translate = useTranslate();
+
+	if ( 'visa' === props.card.card_type ) {
+		logo = <VisaLogo />;
+	} else if ( 'mastercard' === props.card.card_type ) {
+		logo = <MastercardLogo />;
+	} else if ( 'amex' === props.card.card_type ) {
+		logo = <AmexLogo />;
+	}
+
+	const expiry = new Date( props.card.expiry );
+	const month = expiry.getMonth();
+	const year = expiry.getFullYear().toString().substr( 2, 2 );
+
+	return (
+		<div className="payment-method-list__stored-card">
+			<div className="payment-method-list__stored-card-header">
+				<div className="payment-method-list__stored-card-header-left">{ logo }</div>
+
+				<div className="payment-method-list__stored-card-header-right">
+					<Gridicon icon="checkmark" size={ 18 } />
+					<div className="payment-method-list__stored-card-primary">
+						<span className="payment-method-list__stored-card-primary-text">
+							{ translate( 'Primary' ) }
+						</span>
+					</div>
+					<EllipsisMenu className="payment-method-list__stored-card-ellipsis" position="bottom">
+						<PopoverMenuItem>{ translate( 'Make primary' ) }</PopoverMenuItem>
+						<PopoverMenuItem>{ translate( 'Update details' ) }</PopoverMenuItem>
+						<PopoverMenuItem>{ translate( 'Delete card' ) }</PopoverMenuItem>
+					</EllipsisMenu>
```

Comment:
We have changed the designs for this bit, but we can leave it for later since this PR is focused on functionality. But I'm mentioning it here so we can keep it in mind. See: p6TEKc-51M-p2

Reply:
Yeah, I think we can address this in an iteration.